recipes/PeoplesSpeech/ASR/transformer/train.py

```diff
@@ -0,0 +1,458 @@
+            }
+            self.hparams.train_logger.log_stats(
+                stats_meta=epoch_stats,
+                train_stats=self.train_stats,
+                valid_stats=stage_stats,
+            )
+            self.checkpointer.save_and_keep_only(
+                meta={"ACC": stage_stats["ACC"], "epoch": epoch},
+                max_keys=["ACC"],
+            )
+
+        elif stage == sb.Stage.TEST:
+            self.hparams.train_logger.log_stats(
+                stats_meta={"Epoch loaded": self.hparams.epoch_counter.current},
+                test_stats=stage_stats,
+            )
+            if if_main_process():
+                with open(
+                    self.hparams.test_wer_file, "w", encoding="utf-8"
+                ) as w:
+                    self.wer_metric.write_stats(w)
+
+    def on_fit_batch_end(self, batch, outputs, loss, should_step):
+        """At the end of the optimizer step, apply noam annealing."""
+        if should_step:
+            self.hparams.noam_annealing(self.optimizer)
+
+
+def dataio_prepare(hparams):
+    """This function prepares the datasets to be used in the brain class.
+    It also defines the data processing pipeline through user-defined functions.
+    """
+
+    # We load the dataset
+    ds = load_and_concatenate_datasets(
+        hparams["subsets"],
+        hparams["hf_download_folder"],
+    )
+
+    # We must rename the 'id' column because SpeechBrain sampling use this
+    # name for the sampler already, also it's not an id, but an audio_path.
+    train_data = ds[0].rename_column("id", "audio_id")
+    valid_data = ds[1].rename_column("id", "audio_id")
+    test_data = ds[2].rename_column("id", "audio_id")
+
+    # We need to get the full list of durations of all samples to enable
+    # bucketing from the dynamic batch sampler. We do it that way instead
+    # of the usual iterable because the HF dataset ALWAYS open the file
+    # when called, which means that the dynamic sampling needs to read the
+    # 1.5M audio samples from disk.... using a list instead is much master.
+    train_len_list = list(
+        train_data.select_columns("duration_ms")["duration_ms"]
+    )
+    val_len_list = list(valid_data.select_columns("duration_ms")["duration_ms"])
+
+    train_data = sb.dataio.dataset.DynamicItemDataset.from_arrow_dataset(
+        train_data,
+    )
+
+    valid_data = sb.dataio.dataset.DynamicItemDataset.from_arrow_dataset(
+        valid_data,
+    )
+
+    test_data = sb.dataio.dataset.DynamicItemDataset.from_arrow_dataset(
+        test_data,
+    )
+
+    datasets = [train_data, valid_data, test_data]
+
+    # 2. Define audio pipeline:
+    @sb.utils.data_pipeline.takes("audio")
+    @sb.utils.data_pipeline.provides("sig")
+    def audio_pipeline(audio):
+        if audio["sampling_rate"] != 16000:
+            sig = torchaudio.transforms.Resample(
+                audio["sampling_rate"],
+                16000,
+            )(audio["array"])
```

Comment:
shouldn't we have sampling_rate arg in the yaml? Also, dont we have our own interface for resmapling ?

Reply:
Nah and Nah. It should be 16000 anyway. And why use an interface when you can use the native function?